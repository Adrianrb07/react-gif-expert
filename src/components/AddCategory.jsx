import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length <= 1) return;
    onNewCategory(inputValue.trim());
    setInputValue('');
  };

  return (
    <form onSubmit={onSubmit} className="search-form">
      <div className="search-container">
        <input
          type="text"
          placeholder="Buscar gifs"
          value={inputValue}
          onChange={onInputChange}
          className="search-input"
        />
        <button type="submit" className="search-button">
          <i className="fas fa-search"></i>
        </button>
      </div>
    </form>
  );
};
