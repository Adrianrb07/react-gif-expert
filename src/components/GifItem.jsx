import PropTypes from 'prop-types'


export const GifItem = ( { title, url, id } ) => {

  return (
    <div className="card">
        <img className="gif-image"  src={ url } alt={ title } />
        <p>{ title }</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}