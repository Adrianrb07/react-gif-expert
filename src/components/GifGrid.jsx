import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category, onRemoveCategory }) => {

  const { images, isLoading } = useFetchGifs(category);

  return (
    <div className="category-container">
      <div className="category-header">
        <h3 className="category">{category}</h3>
        <div className="tooltip-container">
          <button className="delete-button" onClick={() => onRemoveCategory(category)}>
            <i className="fas fa-trash-alt"></i>
            <span className="tooltip-text">Eliminar categor√≠a</span>
          </button>
        </div>
      </div>
      {isLoading && <p>Cargando...</p>}
      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </div>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
