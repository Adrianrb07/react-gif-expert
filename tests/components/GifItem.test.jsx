import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('Pruebas en <GifItem />', () => { 

    test('debe hacer match con el snapshot', () => {

        const title = 'Goku';
        const url = 'https://localhost/algo.jpg';

        const { container } = render(<GifItem title={title} url={url} />);
        
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar la imagen con la URL y el Alt indicado', () => {

        const title = 'Goku';
        const url = 'https://localhost/algo.jpg';

        render(<GifItem title={title} url={url} />);
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );
        // Otra forma de hacerlo es con desestructuración
        const { src , alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );        
    });
    
    test('debe mostrar el título del componente', () => {

        const title = 'Goku';
        const url = 'https://localhost/algo.jpg';

        render(<GifItem title={title} url={url} />);
        expect( screen.getByText(title) ).toBeTruthy();     
    });

 });