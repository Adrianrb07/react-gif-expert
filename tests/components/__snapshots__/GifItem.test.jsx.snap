// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      class="gif-image"
      src="https://localhost/algo.jpg"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
